src/flb_mp.c

```diff
@@ -350,6 +350,61 @@ void flb_mp_array_header_end(struct flb_mp_map_header *mh)
     flb_mp_set_array_header_size(ptr, mh->entries);
 }
 
+/* TODO: move to mk_list.h ? */
+static inline void list_add_before(struct mk_list *_new,
+                                   struct mk_list *next,
+                                   struct mk_list *head)
+{
+    struct mk_list *prev;
+
+    if (_new == NULL || next == NULL || head == NULL) {
+        return;
+    }
+
+    if ((head->prev == head && head->next == head) /*empty*/||
+        next == head) {
+        mk_list_add(_new, head);
+        return;
+    }
+
+    prev = next->prev;
+    _new->next = next;
+    _new->prev = prev;
+    prev->next = _new;
+    next->prev = _new;
+}
```

Comment:
I added this local function since mk_list API doesn't support below operation3.

1. head `mk_list_init(&head)`
2. head -> val1 `mk_list_add(&val1, &head)`
3. head -> val2 -> val1 (???)


It may be better to move this codes to mk_list.h. However it will take long time. 
Send PR to monkey -> (waiting to merge) -> Send PR to fluent-bit -> (Waiting to merge).

I will send PR to monkey if this patch will work correctly and I will remove this local function.

Reply:
Hah, funny enough I both Hiroshi and I added prepend to monkey a while ago but we still haven't updated monkey in fluent-bit.

Reply:
https://github.com/fluent/fluent-bit/pull/6135 was merged and we can use `mk_list_add_before`.